### PR TITLE
Bump version to 3.1.2

### PR DESCRIPTION
**Pull request by frenchwr**
In addition to bumping to the latest upstream version, this PR does the following:

* Adds a checkbox provider
* Adds GitHub actions
* Switches to using graphics packages from the intel-graphics PPA rather than packages from upstream

### CODE REVIEW

**Review by hector-cao — CHANGES_REQUESTED** [review]
Great work ! Some preliminary comments and questions,

I did not yet review the checkbox provider, perhaps @mckees will go over it before I could, let see

File: .github/testflinger/job-def.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    _run sudo snap install openvino-ai-plugins-gimp --channel latest/edge
+
+    _run sudo snap install checkbox24
+    _put attachments/test/checkbox-openvino-ai-plugins-gimp.snap :/home/ubuntu/checkbox-openvino-ai-plugins-gimp.snap
+    _run sudo snap install --dangerous --classic /home/ubuntu/checkbox-openvino-ai-plugins-gimp.snap
+
+    _put attachments/test/set-device-access.sh :/home/ubuntu/set-device-access.sh
+
+    _run checkbox-openvino-ai-plugins-gimp.install-full-deps
+    _run sudo bash set-device-access.sh
```
[the remark this answers is not in the document]
Comment (frenchwr) [reply .github/testflinger/job-def.yaml]: Good idea - done.

File: .github/testflinger/set-device-access.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+sudo $(basename "${BASH_SOURCE[0]}")
+  "
+  exit 1
+fi
+
+usermod -a -G render ubuntu
+# Newer versions of Ubuntu ship Udev rules which handle this,
+# but 22.04 does not
+chown root:render /dev/accel/accel0
```
[the remark this answers is not in the document]
Comment (frenchwr) [reply .github/testflinger/set-device-access.sh]: Yeah, also a good idea. Done.

File: .github/workflows/integration-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        shell: bash
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          revision=$(snap info openvino-ai-plugins-gimp | grep latest/edge | cut -d "(" -f2 | cut -d ")" -f1)
+          version=$(snap info openvino-ai-plugins-gimp | grep latest/edge | awk '{print $2}')
+          tag_name="${version}-rev${revision}"
+          git tag -a "${tag_name}" -m "Revision ${revision}"
```
[the remark this answers is not in the document]
Comment (frenchwr) [reply .github/workflows/integration-tests.yaml:99]: There may be edge cases where this is possible and this step fails, but in that case we could (i) live with the failure (since it just means the integration test was redundant) or (ii) just delete the tag and re-run this step.

File: .github/workflows/integration-tests.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      - 'checkbox/bin/**'
+      - 'checkbox/checkbox-provider-openvino-ai-plugins-gimp/**'
+      - 'checkbox/snap/**'
+      - '.github/workflows/integration-tests.yaml'
+      - '.github/testflinger/**'
+    branches:
+      - main
+  workflow_dispatch:
+
+concurrency:
```
[the remark this answers is not in the document]
Comment (frenchwr) [reply .github/workflows/integration-tests.yaml:17]: The testflinger submit action handles cancelling the corresponding testflinger job, see: https://github.com/canonical/testflinger/blob/main/.github/actions/submit/action.yaml#L246

File: .github/workflows/integration-tests.yaml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      matrix:
+        device:
+          # Dell XPS 13 9340, Meteor Lake, 22.04
+          - { queue: dell-xps-13-9340-c32267, image: http://10.102.196.9/somerville/Platforms/jellyfish-treecko/FVR_X113/dell-bto-jammy-jellyfish-treecko-X113-20240131-14.iso }
+          # Dell XPS 13 9340, Meteor Lake, 24.04
+          - { queue: dell-xps-13-9340-c34207, image: https://tel-image-cache.canonical.com/oem-share/somerville/releases/noble/oem-24.04a/20240729-53/somerville-noble-oem-24.04a-20240729-53.iso }
+          # Dell Pro Max 14, Arrow Lake, 24.04
+          - { queue: dell-pro-max-14-mc14250-0cf0-c36739, image: https://tel-image-cache.canonical.com/oem-share/somerville/releases/noble/oem-24.04a/20240729-53/somerville-noble-oem-24.04a-20240729-53.iso }
+      fail-fast: false
+    steps:
```
[the remark this answers is not in the document]
Comment (frenchwr) [reply .github/workflows/integration-tests.yaml:61]: Yeah, the three testflinger jobs are run in parallel; however, they do not produce any artifacts that are needed in subsequent steps. They need an artifact (the checkbox provider snap) from the previous step that gets run on a separate runner. You can visualize the steps [here from a successful run](https://github.com/canonical/openvino-ai-plugins-gimp-snap/actions/runs/16483392080) when I had this branch temporarily added to the workflow triggers.

The overall flow is the following, where each bullet blocks the bullet(s) that follow:

* Build checkbox provider snap
* Run integration tests on 3 testflinger devices in parallel, using the checkbox provider snap and the `openvino-ai-plugins-gimp` snap from `latest/edge`
* Push tag for the validated revision

**Comment by frenchwr** [comment]
> Great work ! Some preliminary comments and questions,
> 
> I did not yet review the checkbox provider, perhaps @mckees will go over it before I could, let see

Thanks for the review @hector-cao ! I addressed a few and left comments/responses for the rest. Let me know what you think. The checkbox tests were previously run in the `openvino-toolkit-2404` CI, so they were reviewed in the past.

File: .github/workflows/integration-tests.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (frenchwr) [reply .github/workflows/integration-tests.yaml:99]: Good idea - I'll make that change

File: .github/workflows/integration-tests.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (hector-cao) [reply .github/workflows/integration-tests.yaml:61]: Do we also have the job.yaml file to be generated for each run ?

File: .github/plug-declaration.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
```
[the remark this answers is not in the document]
Comment (hector-cao) [reply .github/plug-declaration.json:1]: nice !

File: .github/workflows/integration-tests.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (frenchwr) [reply .github/workflows/integration-tests.yaml]: Done

File: .github/workflows/integration-tests.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (hector-cao) [reply .github/workflows/integration-tests.yaml:61]: I m wondering what happens if 2 or more runs generate this file at the same time :question:

[the remark this answers is not in the document]
Comment (frenchwr) [reply .github/workflows/integration-tests.yaml:61]: Each of the three jobs are running in parallel on different runners, so there is no danger of overwriting the file. They each have their own copy.

[the remark this answers is not in the document]
Comment (hector-cao) [reply .github/workflows/integration-tests.yaml:61]: Oh I see, each run is on a separate runner, I missed that

**Review by hector-cao — APPROVED** [review]
awesome :rocket:
